docs/client/gas-on-receive.mdx

```diff
@@ -0,0 +1,615 @@
+  try {
+    // Step 1: Get your initial route
+    const originalRoute = await route({
+    amountIn: "1000000", // 1 OSMO
+    sourceAssetChainId: "osmosis-1",
+    sourceAssetDenom: "uosmo",
+    destAssetChainId: "42161", // Arbitrum
+    destAssetDenom: "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1", // WETH
+    smartRelay: true
+  });
+
+  // Step 2: Automatically split into main and gas routes
+  const { mainRoute, gasRoute } = await getRouteWithGasOnReceive({
+    routeResponse: originalRoute,
+    routeRequest: {
+      amountIn: "1000000",
+      sourceAssetChainId: "osmosis-1",
+      sourceAssetDenom: "uosmo",
+      destAssetChainId: "42161",
+      destAssetDenom: "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1",
+      smartRelay: true
+    }
+  });
+
+  // Step 3: Get user addresses for required chains
+  const userAddresses = [
+    { chainId: "osmosis-1", address: "osmo1..." },
+    { chainId: "42161", address: "0x..." }
+  ];
+
+  // Step 4: Execute routes
+  if (gasRoute) {
+    // Execute both routes together
+    await executeMultipleRoutes({
+      route: { mainRoute, feeRoute: gasRoute },
+      userAddresses: {
+        mainRoute: userAddresses,
+        feeRoute: userAddresses // Same addresses for both routes
+      },
+      slippageTolerancePercent: {
+        mainRoute: "1",
+        feeRoute: "10" // Higher tolerance for gas route
+      },
+      getCosmosSigningClient: async (chainId) => {
+        return yourCosmosWallet.getSigningClient(chainId);
+      },
+      getEVMSigningClient: async (chainId) => {
+        return yourEvmWallet.getSigningClient(chainId);
+      },
+      onRouteStatusUpdated: (status) => {
+        console.log("Route status:", status);
+        
+        // Check if gas route failed
+        const gasRouteFailed = status.relatedRoutes?.find(
+          r => r.routeKey === "feeRoute" && r.status === "failed"
+        );
+        
+        if (gasRouteFailed) {
+          console.warn("Gas route failed, but main swap continues");
+        }
+      }
+    });
+  } else {
+    // No gas route needed, execute original route
+    await executeRoute({
+      route: originalRoute,
+      userAddresses,
+      slippageTolerancePercent: "1",
+      getCosmosSigningClient: async (chainId) => {
+        return yourCosmosWallet.getSigningClient(chainId);
+      },
+      getEVMSigningClient: async (chainId) => {
+        return yourEvmWallet.getSigningClient(chainId);
+      }
+    });
+  } catch (error) {
+    console.error("Failed to execute swap:", error);
+    // Handle error appropriately
+  }
+}
+```
+
+### How getRouteWithGasOnReceive Works
+
+The function automatically:
+1. Checks if the destination chain is supported (excludes Solana chains)
+2. Checks source chain support (excludes Ethereum mainnet and Sepolia testnet)
+3. Verifies the destination asset isn't already a fee asset
+4. Calculates appropriate gas amounts:
+   - **Cosmos chains**: $0.10 USD worth of native tokens
```

Comment:
for cosmos chains it's average gas price * 3 but gasPrice is undefined it will fallback to $0.1